### PR TITLE
Source detection with daostarfind

### PR DESCRIPTION
**Pull request by stsci-hack**
This set of changes attempts to replace the segmentation-based centering with a more precise centering algorithm based on DAOStarFinder.  In addition, a couple of other changes were included; namely,

- Default threshold for ACS/SBC data was found to be more useful at a lower value (2.0 instead of 10.0)
- FWHMPSF value for ACS/WFC is actually (based on the ACS Instrument Handbook) 0.13, not 0.075, and this new value was shown to improve the accuracy of the positions as well
- the parameter 'use2dhist' was turned on as it should be more useful with the new more accurate/consistent source positions

A new parameter was added to the interface for 'extract_sources' to allow the calling routine to ask for results with either the old segmentation code or the newer starfind centering code.  This would allow the calling code to use whatever is found to produce the most useful catalog based on the sources in the field of view, with fields containing nearly all extended sources possibly benefiting from the original code.

### CODE REVIEW

File: tests/test_extract.py [hunk 1]
```diff
@@ -1,176 +1,179 @@
...
+
+    ref_loc = ['truth']
+
+    @pytest.mark.parametrize("input_filenames, truth_file",
+                                [('j8ep04lwq','j8ep04lwq_sky.cat'),
+                                 ('J8D806010','j8d806bgq_sky_cat.ecsv'),
+                                 ('IB2V09010', 'ib2v09kzq_sky_cat.ecsv')]
+                            )
+    def XXtest_generate_catalog(self,input_filenames, truth_file):
+        """ Verify whether sources from astrometric catalogs can be extracted from images.
```
Comment (mdlpstsci) [inline tests/test_extract.py]: I will assume this test has been deliberately disabled at this time.

Comment (stsci-hack) [reply tests/test_extract.py]: Actually, this was an oversight from my own runs of these tests.  It has been turned back on in https://github.com/spacetelescope/hlapipeline/pull/43/commits/a1f683ca6072d6466d06009e1b2884cce548cd5e

**Comment by stsci-hack** [comment]
This PR turns on 'use2dhist' in conflict with the changes implemented by Mike for iteratively solving for the tolerance value.  We should NOT merge this PR until it has been decided how to get the two approaches to work together.

**Comment by stsci-hack** [comment]
Continued testing on this PR makes it clear that more work still needs to be done to get this algorithm to behave more robustly.  In addition, more work needs to be done to have the different source identification methods used in different iterations, something the current code would struggle to do.  Therefore, this PR should NOT be merged until/unless these issues are corrected.

**Comment by stsci-hack** [comment]
Commit 366dd8a refines the matching parameters to not be so broad and to successfully use the tweakwcs 2dhist mode more reliably.  In addition, a new parameter was created for the source extraction step to limit the number of sources which gets measured.  This parameter, with a default of 100 sources/chip, sorts the segments by area and only allows the 'nlargest' sources to be measured and added to the source catalog for cross-matching with the reference catalog.  This allows the now very expensive starfind algorithm to work in a managable (but still long) time on images with extremely large numbers of sources (like JC8X16020 at the galactic center).  

In addition, the logic was modified to print out the fit information for all chips regardless of success/failure.

These changes were subsequently used to fit the following datasets to GAIA with improvements in the total rms over what was computed before:
  - JC8X16020 : 16.9069mas
  - J8D806010 : 10.38279mas
  - jbz044010 : 9.93287mas
  - IC8OA5020 : 34.579954mas
Given that it now works reasonably well on WFC3/IR as well as ACS/WFC, this version of the code could be considered for merging if we decide how to resolve the differences between use of 2dhist here compared to the iterative tolerance code from #42 .

**Comment by stsci-hack** [comment]
This PR has been updated with the changes in logic implemented in PR #50 .  

In addition, the cross-matching and fitting operations were merged into a single function that could be called more easily in the top-level calling code.

**Pull request merged**